src/Components/Modal/Modal.js

```diff
@@ -152,29 +158,49 @@ function Modal({
 
     const iconClasses = classNames('mr-3', iconClass);
 
-    return title ? (
-      <Block
-        as="header"
-        justify="between"
-        alignItems="start"
-        padding={['4', '4 5']}
-        className={headerClasses}
-      >
-        <Block className="mr-3">
-          {icon && (
-            <Icon name={icon} size="24" className={iconClasses} />
-          )}
-          {title && (
-            <Heading element="4" responsive={false}>
-              {title}
-            </Heading>
-          )}
+    let headerMarkup = null;
+    const hasTitle = title !== undefined;
+    const hasHeader = header !== undefined;
```

Comment:
maybe use inline `if (header !== undefined)` or `Boolean(header)`

Reply:
My thought was that if header is defined, title is ignored.